authentik/lib/config.py

```diff
@@ -113,6 +137,31 @@ def __init__(self, **kwargs):
                         self.update_from_file(env_file)
         self.update_from_env()
         self.update(self.__config, kwargs)
+        self.check_deprecations()
+
+    def check_deprecations(self):
+        """Warn if any deprecated configuration options are used"""
+
+        def _pop_deprecated_key(current_obj, dot_parts, index):
+            """Recursive function to remove deprecated keys in configuration"""
+            dot_part = dot_parts[index]
+            if index == len(dot_parts) - 1:
+                return current_obj.pop(dot_part)
+            value = _pop_deprecated_key(current_obj[dot_part], dot_parts, index + 1)
+            if not current_obj[dot_part]:
+                current_obj.pop(dot_part)
+            return value
+
+        for deprecation, replacement in DEPRECATIONS.items():
+            if self.get(deprecation, default=UNSET) is not UNSET:
+                self.log(
```

Comment:
We may also want to send notifications to the admins here.

Reply:
How does one send notifications to admins?

Reply:
Using `Notifications.objects.create` ?

Reply:
it would be with `Event.new` with the action set to `configuration_error`, but the config loader purposefully doesn't require any database model to be loaded so I'd prefer to keep it that way (maybe we could have a dynamic import and handle it failing?)